source/App.cpp

```diff
@@ -87,34 +86,42 @@ void App::onInit() {
 	updateMouseSensitivity();			// Update (apply) mouse sensitivity
 	updateSessionDropDown();			// Update the session drop down to remove already completed sessions
 	updateSessionPress();				// Update session to create results file/start collection
+
+	setFrameDuration(m_wallClockTargetDuration, REAL_TIME);
 }
 
 /** Handle then user settings window visibility */
 void App::openUserSettingsWindow() {
-    m_userSettingsMode = true;
-    m_userSettingsWindow->setVisible(m_userSettingsMode);
+	m_userSettingsMode = true;
+	m_userSettingsWindow->setVisible(m_userSettingsMode);
+}
+
+/** Handle then user settings window visibility */
+void App::closeUserSettingsWindow() {
+	m_userSettingsMode = false;
+	m_userSettingsWindow->setVisible(m_userSettingsMode);
 }
 
 /** Update the mouse mode/sensitivity */
 void App::updateMouseSensitivity() {
-    const UserConfig* user = userTable.getCurrentUser();
-    // Converting from mouseDPI (dots/in) and sensitivity (cm/turn) into rad/dot which explains cm->in (2.54) and turn->rad (2*PI) factors
-    // rad/dot = rad/cm * cm/dot = 2PI / (cm/turn) * 2.54 / (dots/in) = (2.54 * 2PI)/ (DPI * cm/360)
-    const double mouseSensitivity = 2.0 * pi() * 2.54 / (user->cmp360 * user->mouseDPI);
-    const shared_ptr<FirstPersonManipulator>& fpm = dynamic_pointer_cast<FirstPersonManipulator>(cameraManipulator());
-    if (m_userSettingsMode) {
-        // Set to 3rd person (i.e. show the mouse cursor)
-        fpm->setMouseMode(FirstPersonManipulator::MouseMode::MOUSE_DIRECT_RIGHT_BUTTON);
-    }
-    else {
-        // Set to first-person mode (i.e. hide the mouse cursor)
-        fpm->setMouseMode(FirstPersonManipulator::MouseMode::MOUSE_DIRECT);
-    }
+	const UserConfig* user = userTable.getCurrentUser();
+	// Converting from mouseDPI (dots/in) and sensitivity (cm/turn) into rad/dot which explains cm->in (2.54) and turn->rad (2*PI) factors
+	// rad/dot = rad/cm * cm/dot = 2PI / (cm/turn) * 2.54 / (dots/in) = (2.54 * 2PI)/ (DPI * cm/360)
+	const double pixelsToRadians = 2.0 * pi() * 2.54 / (user->cmp360 * user->mouseDPI);
+	const shared_ptr<FirstPersonManipulator>& fpm = dynamic_pointer_cast<FirstPersonManipulator>(cameraManipulator());
+	if (m_userSettingsMode) {
+		// Set to 3rd person (i.e. show the mouse cursor)
+		fpm->setMouseMode(FirstPersonManipulator::MouseMode::MOUSE_DIRECT_RIGHT_BUTTON);
+	}
+	else {
+		// Set to first-person mode (i.e. hide the mouse cursor)
+		fpm->setMouseMode(FirstPersonManipulator::MouseMode::MOUSE_DIRECT);
+	}
 	// Control player motion using the experiment config parameter
 	shared_ptr<PlayerEntity> player = scene()->typedEntity<PlayerEntity>("player");
 	if (notNull(player)) {
-		player->mouseSensitivity = (float)mouseSensitivity;
-		player->turnScale = currentTurnScale();
+		player->m_pixelsToRadians = (float)pixelsToRadians;
+		player->turnScale = sessConfig->player.turnScale * userTable.getCurrentUser()->turnScale;
```

Comment:
This is not a replacement for the `App::currentTurnScale()` logic, which includes scaling for when the player is in "scoped" mode. I'm not sure that is an issue given the context here (typically only when the player changes), but I suppose technically the user _could be_ in scoped mode when this happens (causing the loaded turn scale to be incorrect).

I assume this is okay here b/c we assume the user isn't scoped/call `currentTurnScale()` elsewhere, but is there a good reason to avoid calling that function here?

Reply:
oops. looks like this is an integration mistake. should only be changing the first line. thanks!